flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperatorTest.java

```diff
@@ -65,12 +65,13 @@
 import org.apache.flink.util.Preconditions;
 import org.apache.flink.util.TestLogger;
 
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
 import org.hamcrest.Matchers;
 import org.junit.Assert;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.Timeout;
-import org.testcontainers.shaded.com.google.common.collect.Lists;
```

Comment:
How about a checkstyle rule to forbid imports from org.testcontainers.shaded?

Reply:
Yes, I think it makes sense to do so. Do you want to add this rule to this PR?

Reply:
Yes, let's do it here that way adding this checkstyle will have a bit more of a context. (checkstyle doesn't have to be backported)

Reply:
I pushed the separate commit with checkstyle changes